Formula/asimov.rb

```diff
@@ -0,0 +1,53 @@
+class Asimov < Formula
+  desc "Automatically exclude development dependencies from Time Machine backups"
+  homepage "https://github.com/stevegrunwell/asimov"
+  url "https://github.com/stevegrunwell/asimov/archive/v0.2.0.tar.gz"
+  sha256 "2efb456975af066a17f928787062522de06c14eb322b2d133a8bc3a764cc5376"
+  revision 0
+
+  # Setup HEAD support (install with --HEAD)
+  head "https://github.com/stevegrunwell/asimov.git", :branch => "develop"
+
+  # No bottling necessary
+  bottle :unneeded
+
+  # Define the installation steps
+  def install
+    # Install the application
+    bin.install buildpath/"asimov"
+  end
+
+  # Define launch agent plist options (launch automatically, but also allow manual startup)
+  plist_options :startup => true, :manual => "asimov"
+
+  # Define the agent plist contents
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+          <dict>
+              <key>Label</key>
+              <string>#{plist_name}</string>
+              <key>Program</key>
+              <string>#{opt_bin}/asimov</string>
+              <key>StartInterval</key>
+              <!-- 24 hours = 60 * 60 * 24 -->
+              <integer>86400</integer>
+          </dict>
+      </plist>
+    EOS
+  end
+
+  # Define tests to run
+  test do
+    # Test that the binary exists
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.